packages/remix-react/components.tsx

```diff
@@ -1248,16 +1248,29 @@ export interface SubmitFunction {
  *
  * @see https://remix.run/api/remix#usesubmit
  */
-export function useSubmit(): SubmitFunction {
-  return useSubmitImpl();
+export function useSubmit(options?: SubmitOptions): SubmitFunction {
+  let key = undefined; // no key for global submit
```

Comment:
I thought about making `key` a part of the options object in `useSubmitImpl` instead of accepting it as a first param, but decided to not cause a breaking change to `useSubmitImpl` because it's exported and it's possible people are using it. I actually don't hate this though so maybe it's fine?

Reply:
Since I think we can safely remove the export (as it was never intended to be a public API), I think we can make `key` part of the `submitOptions`

Reply:
I actually kinda like that `submitOptions` is a separate argument, even though we have this funny `key = undefined` thing. But I don't care enough.